tensorflow/contrib/opt/python/training/model_average_optimizer.py

```diff
@@ -0,0 +1,693 @@
+        init_tokens = self._sync_token_queue.enqueue_many((tokens,))
+    else:
+      init_tokens = control_flow_ops.no_op(name="no_init_tokens")
+
+    return init_tokens
+
+  def make_ma_run_hook(self):
+    local_vars = variables.trainable_variables()
+    global_vars = ops.get_collection_ref('global_model')
+    v_vars = ops.get_collection_ref('block_grads')
+    nesterov_vars = ops.get_collection_ref('nesterov')
+
+    if self._use_nesterov:
+      self._apply_refresh_local_vars_op = self.apply_refresh_local_vars(
+                                                                  local_vars,
+                                                                  nesterov_vars)
+      local_and_init_vars = list(zip(local_vars, nesterov_vars))
+      self._apply_init_nesterov_vars_op = self.apply_refresh_local_vars(
+                                                                  nesterov_vars,
+                                                                  global_vars)
+    else:
+      self._apply_refresh_local_vars_op = self.apply_refresh_local_vars(
+                                                                  local_vars,
+                                                                  global_vars)
+      local_and_init_vars = list(zip(local_vars, global_vars))
+      self._apply_init_nesterov_vars_op = None
+
+    self._apply_ma_op = self.apply_model_average(local_and_init_vars,
+                                           global_vars,
+                                           v_vars,
+                                           nesterov_vars,
+                                           global_step=self._super_global_step)
+    self._ma_run_hook = True
+
+    return self._ModelAverageHook(self, self._apply_refresh_local_vars_op,
+                                    self._apply_init_nesterov_vars_op,
+                                    self._apply_ma_op,
+                                    self._interval_steps,
+                                    self._super_global_step)
+
+  def make_session_run_hook(self, is_chief, num_tokens=0):
+    """Creates a hook to handle ReplicasHook ops such as initialization."""
+    if self._ma_run_hook is False:
+      raise ValueError("make_session_run_hook Should be "
+                       "called after make_ma_run_hook.")
+
+    if is_chief:
+      return self._ReplicasHook(self.chief_init_op,
+                                    self.ready_for_local_init_op,
+                                    self.get_chief_queue_runner(),
+                                    self.get_init_tokens_op(num_tokens))
+
+    return self._ReplicasHook(self.local_step_init_op,
+                                  self.ready_for_local_init_op, None, None)
+
+  class _ModelAverageHook(session_run_hook.SessionRunHook):
+    def __init__(self, parent, local_refresh_op, nesterov_init_op, apply_ma_op,
+                 interval_steps, super_global_step):
+      self._local_refresh_op = local_refresh_op
```

Comment:
Use the same naming throughout. If I'm reading this correctly, elsewhere this is called _apply_refresh_local_vars_op